### PR TITLE
Re-design Habit Model

### PR DESCRIPTION
**Pull request by MacMalainey**
Redesign the habit model to use database service
Old HabitModel still exists for legacy sake for now

## Description of Changes

- Re-implement Habit Model
- Allow pushing model changes to database
- Allow getting model data from database
- Allow creating new habit model and adding to database

## Checklist

- [ ] Automated tests
- [x] Screenshots included (if necessary)
- [x] Code is well commented

Closes #ISSUE_ID

### CODE REVIEW

**Review by Docker-J — APPROVED** [review]

File: src/app/src/main/java/com/cmput301f21t09/budgetprojectname/models/IHabitModel.java [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+
+/**
+ * Interface class for user habits
+ */
+public interface IHabitModel {
+
+    /**
+     * Habit maximum title length
+     */
+    int MAX_TITLE_LENGTH = 20;
```
Comment (j-vent) [inline src/app/src/main/java/com/cmput301f21t09/budgetprojectname/models/IHabitModel.java:13]: make values final

**Review by j-vent — APPROVED** [review]
I think this made the project more complicated than necessary

File: src/app/src/main/java/com/cmput301f21t09/budgetprojectname/models/IHabitModel.java (hunk 1 above)
Comment (MacMalainey) [reply src/app/src/main/java/com/cmput301f21t09/budgetprojectname/models/IHabitModel.java:13]: static interface values are already final

**Comment by MacMalainey** [comment]
> I think this made the project more complicated than necessary

That may absolutely be true.

**Review by cheetiantey — APPROVED** [review]

File: src/app/src/main/java/com/cmput301f21t09/budgetprojectname/models/HabitModel.java [hunk 2]
```diff
@@ -0,0 +1,224 @@
...
+    /**
+     * Set the start date of the habit
+     * @param date to set
+     */
+    public void setStartDate(Date date) {
+        this.startDate = date;
+    }
+
+    @Override
+    public Date getLastCompleted() {
```
Comment (loravocado) [inline src/app/src/main/java/com/cmput301f21t09/budgetprojectname/models/HabitModel.java:169]: Do you need javadoc for this?

**Pull request merged**